### PR TITLE
VFB-185: Log changes when a client is added or edited

### PR DESCRIPTION
**Pull request by minalui**
## What's changed
- Removed family_member_id from audit log table
- Updated supabase
- Add audit logging for add and edit clients

## Screenshots / Videos
| Before     | After      |
|------------|------------|
| paste_here | paste_here |

## Checklist
- [x] The ticket is up-to-date - Please document any deviations from the original approach if there is any.
- [x] I have documented the testing steps for QA
- [x] I have self-reviewed this PR
- [x] Make sure you've verified it works via `npm run dev`
- [x] Make sure you've verified it works via `npm run build` and `npm run start`
- [x] Make sure you've fixed all linting problems with `npm run lint_fix`
- [x] Make sure you've tested via `npm run test:e2e`

If you have made any changes to the database...
  - [x] The migration files are up-to-date with my final set up (`npx supabase db diff -f <name_of_migration>` should create nothing at this point)
  - [x] I have updated the typescript definitions for the database with `db:local:generate_types`
  - [x] I have modified the seed in `supabase/seed/seed.mts` if appropriate
  - [x] If I have modified the seed, I have also generated the seed with `npm run db:generate_seed` 
  - [x] With my final set up, I can run `npm run dev:reset_supabase` without any errors.
  - Does this require resetting the deployed database? - YES (remove as appropriate)
  - [x] I have checked that migration can happen successfully without resetting the database, doing the following.
    - Make sure you have rebased your branch onto dev or merged dev into your branch.
    - Checkout the dev branch
    - Run `npm run post_checkout` to reset the database, including the seed data.
    - Checkout your branch
    - Run `npx supabase migrations up --include-all --local` to run all outstanding migration files
    - Check that the resulting database is what you expect.

### CODE REVIEW

**Comment by minalui** [comment]
I think because I updated supabase its different to the checks - should I un-update it?

File: supabase/migrations/20240409155608_remove_family_member_id_column_from_audit_log_table.sql
Comment (rickaw) [inline supabase/migrations/20240409155608_remove_family_member_id_column_from_audit_log_table.sql:1]: Could you update the ticket description to reflect this please?
https://softwiretech.atlassian.net/browse/VFB-77

**Review by rickaw — APPROVED** [review]
Nice!

**Comment by rickaw** [comment]
> I think because I updated supabase its different to the checks - should I un-update it?

Could you update `.github/workflows/verify-generated-database-types.yml` to use version `1.153.4` instead of `1.145.4` please?

**Pull request merged**